### DISCUSSION TITLE
a way to blacklist or whitelist algorithms

### DISCUSSION THREAD

**Discussion by tomholub** [openpgpjs/openpgpjs#1218] [Ideas]
We'd like to have a way, either through whitelisting or blacklisting, to avoid using algorithms that are not so secure.

One example, when encrypting a message for a public key, we'd rather see an error then see the message encrypted with TripleDES.

Posting here to see if it would make sense to include this kind of option in the library.

**Comment by twiss** [comment]
Hey :wave: Yeah, I think it could make sense, but I think there's another reasoning we could use to never use TripleDES, even without a config option:

[RFC4880bis says](https://tools.ietf.org/html/draft-ietf-openpgp-rfc4880bis-08#section-9.3) that "implementations MUST implement AES-128", and regarding [Symmetric Algorithm Preferences](https://tools.ietf.org/html/draft-ietf-openpgp-rfc4880bis-08#section-14.2), that "since AES-128 is the MUST-implement algorithm, if it is not explicitly in the list, it is tacitly at the end". Even though it's not a final RFC yet, I think it's reasonable to assume that all implementations already conform to this, anyway. We already take the liberty, based on this, to make AES-128 the default if there is no overlap between the Preferred Symmetric Algorithms of the public keys.

I think we could take this further by saying: since all public keys' preferences implicitly include AES-128, we can always use that algorithm, _unless_ there is another algorithm that is preferred by all the keys that is _at least_ as strong as AES-128, perhaps by block size? We do something like that in [`getPreferredHashAlgo`](https://github.com/openpgpjs/openpgpjs/blob/v5/src/key/helper.js#L134) currently (by hash size), so I think it would be reasonable to do something similar in [`getPreferredAlgo`](https://github.com/openpgpjs/openpgpjs/blob/v5/src/key/helper.js#L170).

**Comment by tomholub** [comment]
That would be nice - a way to address the issue without having to create special cases :+1:

**Reply by larabr** [reply]
FYI this was implemented in #1262 . The `config` can be changed to decide which algorithm should be used for encryption (see PR for more details).
